lib/Runtime/Library/JavascriptPromise.cpp

```diff
@@ -201,16 +201,18 @@ namespace Js
         remainingElementsWrapper->remainingElements = 1;
 
         JavascriptExceptionObject* exception = nullptr;
-
+        RecyclableObject* iterator = nullptr;
+        bool shouldCallReturn = false;
         try
         {
             // 4. Let iterator be GetIterator(iterable).
-            RecyclableObject* iterator = JavascriptOperators::GetIterator(iterable, scriptContext);
+            iterator = JavascriptOperators::GetIterator(iterable, scriptContext);
             Var next;
             values = library->CreateArray(0);
 
             while (JavascriptOperators::IteratorStepAndValue(iterator, scriptContext, &next))
```

Comment:
I think it would be fragile to leave it as is.  I see the exception if there was one is used afterwards.  Either the while loop can be changed within the try/catch here and it should work the same, or `DoIteratorStepAndValue` can be modified to optionally return the exception instead of rethrowing.  Either way I think that would be cleaner and safer than leaving it as is.

---
In reply to: [66875446](https://github.com/Microsoft/ChakraCore/pull/1107#discussion_r66875446) [](ancestors = 66875446)